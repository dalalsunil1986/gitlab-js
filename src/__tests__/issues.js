jest.mock('../utils/api', () => jest.fn());
jest.mock('extra-log', () => ({ msg: jest.fn() }));

const issues = require('../issues');
const api = require('../utils/api');

describe('issues', () => {
  beforeEach(() => {
    api.mockReset();
    api.mockImplementation((...params) => [...params]);
  });


  it('list', async () => {
    const responseUndefined = await issues.list();
    expect(responseUndefined).toBe(undefined);
    expect(api).toHaveBeenCalledTimes(0);

    const response = await issues.list('projectId');
    expect(response).toMatchSnapshot();
    expect(api).toHaveBeenCalledTimes(1);
    expect(api.mock.calls).toMatchSnapshot();
  });


  it('report', async () => {
    const responseUndefined = await issues.list();
    expect(responseUndefined).toBe(undefined);
    expect(api).toHaveBeenCalledTimes(0);

    const response = await issues.report('projectId', {
      title: 'title',
      description: 'description',
    });

    expect(response).toMatchSnapshot();
    expect(api).toHaveBeenCalledTimes(1);
    expect(api.mock.calls).toMatchSnapshot();
  });


  it('delete', async () => {
    const responseUndefined = await issues.delete();
    expect(responseUndefined).toBe(undefined);
    expect(api).toHaveBeenCalledTimes(0);

    const response = await issues.delete('projectId', 'issueId');

    expect(response).toMatchSnapshot();
    expect(api).toHaveBeenCalledTimes(1);
    expect(api.mock.calls).toMatchSnapshot();
  });


  it('addComment', async () => {
    const responseUndefined = await issues.delete();
    expect(responseUndefined).toBe(undefined);
    expect(api).toHaveBeenCalledTimes(0);

    const responseUndefined2 = await issues.delete('projectId');
    expect(responseUndefined2).toBe(undefined);
    expect(api).toHaveBeenCalledTimes(0);

    const response = await issues.addComment('projectId', 'issueId', 'addComment');

    expect(response).toMatchSnapshot();
    expect(api).toHaveBeenCalledTimes(1);
    expect(api.mock.calls).toMatchSnapshot();
  });
});
