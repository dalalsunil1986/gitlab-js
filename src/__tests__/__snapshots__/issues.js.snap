// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issues addComment 1`] = `
Array [
  "/projects/projectId/issues/issueId/notes",
  Object {
    "body": "addComment",
  },
  Object {
    "method": "POST",
  },
]
`;

exports[`issues addComment 2`] = `
Array [
  Array [
    "/projects/projectId/issues/issueId/notes",
    Object {
      "body": "addComment",
    },
    Object {
      "method": "POST",
    },
  ],
]
`;

exports[`issues delete 1`] = `
Array [
  "/projects/projectId/issues/issueId",
  Object {},
  Object {
    "method": "DELETE",
  },
]
`;

exports[`issues delete 2`] = `
Array [
  Array [
    "/projects/projectId/issues/issueId",
    Object {},
    Object {
      "method": "DELETE",
    },
  ],
]
`;

exports[`issues list 1`] = `
Array [
  "/projects/projectId/issues",
  Object {
    "simple": true,
  },
]
`;

exports[`issues list 2`] = `
Array [
  Array [
    "/projects/projectId/issues",
    Object {
      "simple": true,
    },
  ],
]
`;

exports[`issues report 1`] = `
Array [
  "/projects/projectId/issues",
  Object {
    "description": "description",
    "id": "projectId",
    "title": "title",
  },
  Object {
    "method": "POST",
  },
]
`;

exports[`issues report 2`] = `
Array [
  Array [
    "/projects/projectId/issues",
    Object {
      "description": "description",
      "id": "projectId",
      "title": "title",
    },
    Object {
      "method": "POST",
    },
  ],
]
`;
